extensions/ecc/circuit/src/weierstrass_chip/curves.rs

```diff
@@ -0,0 +1,382 @@
+use halo2curves_axiom::ff::PrimeField;
+use num_bigint::BigUint;
+use num_traits::{FromPrimitive, Num};
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum CurveType {
+    K256 = 0,
+    P256 = 1,
+    BN254 = 2,
+    BLS12_381 = 3,
+}
+
+const K256_A: i64 = 0;
+const P256_A: i64 = -3;
+const BN254_A: i64 = 0;
+const BLS12_381_A: i64 = 0;
+
+fn get_modulus_as_bigint<F: PrimeField>() -> BigUint {
+    BigUint::from_str_radix(F::MODULUS.trim_start_matches("0x"), 16).unwrap()
+}
+
+pub fn get_curve_type_from_modulus(modulus: &BigUint) -> Option<CurveType> {
```

Comment:
isn't it technically unsafe to use this instead of `get_curve_type`?

Reply:
the addition formula just depends on the coordinate field so we don't care about the curve parameters a and b here. this technically is returning a field type. maybe i should just add a separate type for that